In [2]:
!pip install pandas
!pip install numpy
!pip install matplotlib




In [3]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

Abrindo o arquivo .csv

In [4]:
df = pd.read_csv('mkt_data.csv')

In [5]:
# Exibindo o arquivo
df

Unnamed: 0   Income  Kidhome  Teenhome  Recency  MntWines  MntFruits  \
0              0  58138.0        0         0       58       635         88   
1              1  46344.0        1         1       38        11          1   
2              2  71613.0        0         0       26       426         49   
3              3  26646.0        1         0       26        11          4   
4              4  58293.0        1         0       94       173         43   
...          ...      ...      ...       ...      ...       ...        ...   
2200        2200  61223.0        0         1       46       709         43   
2201        2201  64014.0        2         1       56       406          0   
2202        2202  56981.0        0         0       91       908         48   
2203        2203  69245.0        0         1        8       428         30   
2204        2204  52869.0        1         1       40        84          3   

      MntMeatProducts  MntFishProducts  MntSweetProducts  ...  \
0                 546              172                88  ...   
1                   6                2                 1  ...   
2                 127              111                21  ...   
3                  20               10                 3  ...   
4                 118               46                27  ...   
...               ...              ...               ...  ...   
2200              182               42               118  ...   
2201               30                0                 0  ...   
2202              217               32                12  ...   
2203              214               80                30  ...   
2204               61                2                 1  ...   

      education_Graduation  education_Master  education_PhD  MntTotal  \
0                      3.0               NaN            NaN      1529   
1                      3.0               NaN            NaN        21   
2                      3.0               NaN            NaN       734   
3                      3.0               NaN            NaN        48   
4                      NaN               NaN            5.0       407   
...                    ...               ...            ...       ...   
2200                   3.0               NaN            NaN      1094   
2201                   NaN               NaN            5.0       436   
2202                   3.0               NaN            NaN      1217   
2203                   NaN               4.0            NaN       782   
2204                   NaN               NaN            5.0       151   

      MntRegularProds  AcceptedCmpOverall  marital_status  education_level  \
0                1441                   0          Single       Graduation   
1                  15                   0          Single       Graduation   
2                 692                   0        Together       Graduation   
3                  43                   0        Together       Graduation   
4                 392                   0         Married              PhD   
...               ...                 ...             ...              ...   
2200              847                   0         Married       Graduation   
2201              428                   1        Together              PhD   
2202             1193                   1        Divorced       Graduation   
2203              721                   0        Together           Master   
2204              130                   0         Married              PhD   

      kids  expenses  
0        0      1529  
1        2        21  
2        0       734  
3        1        48  
4        1       407  
...    ...       ...  
2200     1      1094  
2201     3       436  
2202     0      1217  
2203     1       782  
2204     2       151  

[2205 rows x 44 columns]

In [6]:
# Exibindo linhas e colunas
df.shape

(2205, 44)

In [7]:
# mostrando os tipos de colunas
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2205 entries, 0 to 2204
Data columns (total 44 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Unnamed: 0            2205 non-null   int64  
 1   Income                2205 non-null   float64
 2   Kidhome               2205 non-null   int64  
 3   Teenhome              2205 non-null   int64  
 4   Recency               2205 non-null   int64  
 5   MntWines              2205 non-null   int64  
 6   MntFruits             2205 non-null   int64  
 7   MntMeatProducts       2205 non-null   int64  
 8   MntFishProducts       2205 non-null   int64  
 9   MntSweetProducts      2205 non-null   int64  
 10  MntGoldProds          2205 non-null   int64  
 11  NumDealsPurchases     2205 non-null   int64  
 12  NumWebPurchases       2205 non-null   int64  
 13  NumCatalogPurchases   2205 non-null   int64  
 14  NumStorePurchases     2205 non-null   int64  
 15  NumWebVisitsMonth    

In [ ]:
# Selecionando apenas as numéricas: aqui colocamos todos os formatos que o Python reconhece como numérico.
numerics = ['int16', 'int32', 'int64', 'float16', 'float32', 'float64']

# .select_dtypes seleciona no dataframe os tipos de dados passados no parâmetro, enquanto o .columns vai mostrar o nome das colunas que possuem aquele tipo de dado.
df.select_dtypes(include=numerics).columns

Index(['Unnamed: 0', 'Income', 'Kidhome', 'Teenhome', 'Recency', 'MntWines',
       'MntFruits', 'MntMeatProducts', 'MntFishProducts', 'MntSweetProducts',
       'MntGoldProds', 'NumDealsPurchases', 'NumWebPurchases',
       'NumCatalogPurchases', 'NumStorePurchases', 'NumWebVisitsMonth',
       'AcceptedCmp3', 'AcceptedCmp4', 'AcceptedCmp5', 'AcceptedCmp1',
       'AcceptedCmp2', 'Complain', 'Z_CostContact', 'Z_Revenue', 'Response',
       'Age', 'Customer_Days', 'marital_Divorced', 'marital_Married',
       'marital_Single', 'marital_Together', 'marital_Widow',
       'education_2n Cycle', 'education_Basic', 'education_Graduation',
       'education_Master', 'education_PhD', 'MntTotal', 'MntRegularProds',
       'AcceptedCmpOverall', 'kids', 'expenses'],
      dtype='object')

In [9]:
# Verificando duplicados no dataset e removendo se caso existir.

# Somando os duplicados do dataset: verificando a soma total de dados nulos
df.duplicated().sum()

np.int64(0)

In [10]:
# outra possibilidade é verificar acerca da existência, aqui você apenas consegue verificar se os duplicados existem, mas não quanto são:
df.duplicated().any()

np.False_

OBSERVAÇÃO:
Observando que há uma coluna no Dataframe chamada "Unnamed: 0", ao rodar o código acima, estamos considerando os valores nela contidos e os dados não apresentarão valores duplicados, uma vez que há valores distintos para cada uma destas linhas. Logo, mesmo que hajam dados duplicados, ao considerar esta coluna, o código não reconhecerá como linhas repetidas.

Como esta coluna em si, não traz nenhuma informação pertinente, por ser índices do dataframe (observa-se que os valores são repetidos na coluna e no índice) você pode eliminar esta coluna, e seguir com as análises.

Para eliminar a coluna "Unnamed: 0", basta seguir estas etapas:

In [16]:
# Eliminando a coluna "Unnamed: 0"
d = df.drop('Unnamed: 0', axis=1)

In [17]:
# Verificando a quantidade de duplicados
df.duplicated().sum()

np.int64(0)

In [18]:
# Removendo dados duplicados
df = df.drop_duplicates()

In [20]:
# Somando os nulos da base
df.isnull().sum()

Unnamed: 0                 0
Income                     0
Kidhome                    0
Teenhome                   0
Recency                    0
MntWines                   0
MntFruits                  0
MntMeatProducts            0
MntFishProducts            0
MntSweetProducts           0
MntGoldProds               0
NumDealsPurchases          0
NumWebPurchases            0
NumCatalogPurchases        0
NumStorePurchases          0
NumWebVisitsMonth          0
AcceptedCmp3               0
AcceptedCmp4               0
AcceptedCmp5               0
AcceptedCmp1               0
AcceptedCmp2               0
Complain                   0
Z_CostContact              0
Z_Revenue                  0
Response                   0
Age                        0
Customer_Days              0
marital_Divorced        1975
marital_Married         1351
marital_Single          1728
marital_Together        1637
marital_Widow           2129
education_2n Cycle      2007
education_Basic         2151
education_Grad

Temos nulos na base, nas colunas:

marital_Divorced
marital_Married
marital_Single
marital_Together
marital_Widow
education_2n Cycle
education_Basic
education_Graduation
education_Master
education_PhD

Os nulos indicados acima podem significar que a pessoa não tem aquela determinada característica. Essas colunas só tem nulos ou um outro determinado valor. Por exemplo:

In [21]:
# Contando as ocorrências de um valor na coluna education_PhD: o .value_counts() conta apenas os registros não nulos.
df.education_PhD.value_counts()

education_PhD
5.0    476
Name: count, dtype: int64

A coluna education_PhD só tem o número 5.0, que aparece 476 vezes. O restante é nulo. Isso é um ótimo indício que o nulo significa que a pessoa não tem aquela determinada variável. No caso acima, ela não tem PHD

Portanto, essas colunas podem se tornar um booleano - 1 se fez e 0 se não fez (nulo)

In [27]:
# Selecionando colunas com nulos
colunas_com_nulos = ['marital_Divorced'
, 'marital_Married'
, 'marital_Single'
, 'marital_Together'
, 'marital_Widow'
, 'education_2n Cycle'
, 'education_Basic'
, 'education_Graduation'
, 'education_Master'
, 'education_PhD']

In [28]:
# Substituindo por booleano

for item in colunas_com_nulos:
    df['booleano'+str(item)] = np.where(df[item].isnull(), 0, 1) # se for nulo, substitui por 0, caso contrário substitui por 1

In [29]:
# Observando as ultimas colunas
df

Unnamed: 0   Income  Kidhome  Teenhome  Recency  MntWines  MntFruits  \
0              0  58138.0        0         0       58       635         88   
1              1  46344.0        1         1       38        11          1   
2              2  71613.0        0         0       26       426         49   
3              3  26646.0        1         0       26        11          4   
4              4  58293.0        1         0       94       173         43   
...          ...      ...      ...       ...      ...       ...        ...   
2200        2200  61223.0        0         1       46       709         43   
2201        2201  64014.0        2         1       56       406          0   
2202        2202  56981.0        0         0       91       908         48   
2203        2203  69245.0        0         1        8       428         30   
2204        2204  52869.0        1         1       40        84          3   

      MntMeatProducts  MntFishProducts  MntSweetProducts  ...  \
0                 546              172                88  ...   
1                   6                2                 1  ...   
2                 127              111                21  ...   
3                  20               10                 3  ...   
4                 118               46                27  ...   
...               ...              ...               ...  ...   
2200              182               42               118  ...   
2201               30                0                 0  ...   
2202              217               32                12  ...   
2203              214               80                30  ...   
2204               61                2                 1  ...   

      booleanomarital_Divorced  booleanomarital_Married  \
0                            0                        0   
1                            0                        0   
2                            0                        0   
3                            0                        0   
4                            0                        1   
...                        ...                      ...   
2200                         0                        1   
2201                         0                        0   
2202                         1                        0   
2203                         0                        0   
2204                         0                        1   

      booleanomarital_Single  booleanomarital_Together  booleanomarital_Widow  \
0                          1                         0                      0   
1                          1                         0                      0   
2                          0                         1                      0   
3                          0                         1                      0   
4                          0                         0                      0   
...                      ...                       ...                    ...   
2200                       0                         0                      0   
2201                       0                         1                      0   
2202                       0                         0                      0   
2203                       0                         1                      0   
2204                       0                         0                      0   

      booleanoeducation_2n Cycle  booleanoeducation_Basic  \
0                              0                        0   
1                              0                        0   
2                              0                        0   
3                              0                        0   
4                              0                        0   
...                          ...                      ...   
2200                           0                        0   
2201                           0                        0   
2202                           0                        0

Descobrindo a média, mediana, 25 percentil, 75 percentil, mínimo e máximo das colunas income, kids e expenses

In [31]:
# Utilizando o describe() para mostrar as principais estatísticas das colunas numéricas de interesse
df[['kids', 'Income', 'expenses']].describe()

kids         Income     expenses
count  2205.000000    2205.000000  2205.000000
mean      0.948753   51622.094785   562.764626
std       0.749231   20713.063826   575.936911
min       0.000000    1730.000000     4.000000
25%       0.000000   35196.000000    56.000000
50%       1.000000   51287.000000   343.000000
75%       1.000000   68281.000000   964.000000
max       3.000000  113734.000000  2491.000000